### PR TITLE
Add a watcher for the share panel of a space

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
We've added a watcher for the share panel of a space to ensure seamless navigation via the share indicator.

## Related Issue

- See comment: https://github.com/owncloud/web/pull/6455#discussion_r820598017

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/src/components/SideBar/Shares/SpaceShares.vue [hunk 1]
```diff
@@ -87,6 +87,16 @@ export default {
...
+      handler: function (newItem, oldItem) {
+        if (oldItem !== newItem) {
+          this.loadSharesTask.perform(this)
+        }
+      },
+      immediate: true
+    }
+  },
   mounted() {
     this.loadSharesTask.perform(this)
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/Shares/SpaceShares.vue:101]: with `immediate: true` you should be able to remove the loading task from the mounted hook (and with that the delete the mounted hook entirely).

Comment (JammingBen) [reply packages/web-app-files/src/components/SideBar/Shares/SpaceShares.vue]: Indeed, thx!

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/23348/66/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:94

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23348/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:84

**Pull request merged**